pkg/vex/controller/reconciler_test.go

```
@@ -0,0 +1,141 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func TestVEXSourceReconciler_Reconcile(t *testing.T) {
+	// Setup a fake HTTP server returning valid OpenVEX JSON
+	mockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		w.Write([]byte(`{
```

golangci-lint (2.12.2)

[error] Error return value of `w.Write` is not checked

(errcheck)